src/libnnstreamer-edge/nnstreamer-edge-common.c

```diff
@@ -599,3 +601,37 @@ nns_edge_data_get_info (nns_edge_data_h data_h, const char *key, char **value)
 
   return NNS_EDGE_ERROR_NONE;
 }
+
+/**
+ * @brief Get IP address
+ * @note Caller should release newly allocated string using nns_edge_free ().
+ */
+gchar *
+nns_edge_get_ip_address (void)
+{
+  struct ifaddrs *addrs;
+  gchar *ret = NULL;
+
+  getifaddrs (&addrs);
+  while (addrs) {
+    if (addrs->ifa_addr && addrs->ifa_addr->sa_family == AF_INET) {
+      struct sockaddr_in *pAddr = (struct sockaddr_in *) addrs->ifa_addr;
+
+      if (NULL != strstr (addrs->ifa_name, "en") ||
+          NULL != strstr (addrs->ifa_name, "et")) {
+        g_free (ret);
+        ret = g_strdup (inet_ntoa (pAddr->sin_addr));
+        break;
+      }
+    }
+    addrs = addrs->ifa_next;
+  }
+  while ((addrs = addrs->ifa_next));
```

Comment:
Refer to https://github.com/nnstreamer/nnstreamer/pull/3859/files

This may incur segmentation fault (accessing nullptr).

CC: @jaeyun-jung